PHP-ARRAYS/exercicio_php_b11.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+$estoque = [];
+
+function adicionarProduto(&$estoque, $codigo, $nome, $tamanho, $cor, $quantidade)
+{
+    $estoque[$codigo] = [
+        'nome' => $nome,
+        'tamanho'=> $tamanho,
+        'cor'=> $cor,
+        'quantidade' => $quantidade
+    ];
```

Comment:
Gostei muito do seu raciocínio em colocar o código como indice do array. Funciona aqui e nas funcçoes abaixo ficou mais limpo o código. Porém, se esses dados dos produtos vierem de um banco de dados (que essa era a ideia, simular isso) o código não seria a chave principal do array. No banco de dados seria o ID (que sé um indice numerico sequencial), então esse Codigo do Produto (que seria como o SKU, que pode ser alfanumérico) ficaria como um valor, assim como nome, tamanho, etc.
O motivo é que esse código (SKU) precisa ser passível de alteração. Se ele for tratado como chave primária, não tem como alterar depois. Chaves primárias não alteram de valor e são únicos.
Quando estivermos tratando de banco de dados vai entender melhor.
A princípio pode deixar como está e não precisa alterar. Mas é bom você já ir sabendo desses detalhes.